core/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -42,27 +45,43 @@
     private AggregatedDfs aggregatedDfs;
```

Comment:
I don't know how often this is used but maybe we should use inheritance here to differentiate between profiled and unprofiled? That way we simply override `createNormalizedWeight` and `createWeight` without any mutable profiler? but I might not see all the implications this will have when we create the seracher etc?